From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xVorttex <sigamatute@gmail.com>
Date: Mon, 16 Feb 2026 05:37:55 +0200
Subject: [PATCH] Unused stuff


diff --git a/io/papermc/paper/entity/activation/ActivationRange.java b/io/papermc/paper/entity/activation/ActivationRange.java
index 45b17a4fbb7f4941f23f5bf4983f9eccf425b73f..08553d0e47a76e7cbcc9649c0c0dc23f5f81e162 100644
--- a/io/papermc/paper/entity/activation/ActivationRange.java
+++ b/io/papermc/paper/entity/activation/ActivationRange.java
@@ -161,8 +161,6 @@ public final class ActivationRange {
                 continue;
             }
 
-            if (!player.level().purpurConfig.idleTimeoutTickNearbyEntities && player.isAfk()) continue; // Purpur - AFK API
-
             final int worldHeight = world.getHeight();
             ActivationRange.maxBB = player.getBoundingBox().inflate(maxRange, worldHeight, maxRange);
             ActivationType.MISC.boundingBox = player.getBoundingBox().inflate(miscActivationRange, worldHeight, miscActivationRange);
diff --git a/net/minecraft/commands/Commands.java b/net/minecraft/commands/Commands.java
index 1b49392ed30d69e74732187eb70489ee77a2a89c..57fdda94ce810f8309d2445966f1691d082d9980 100644
--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -15,15 +15,6 @@ import com.mojang.brigadier.tree.ArgumentCommandNode;
 import com.mojang.brigadier.tree.CommandNode;
 import com.mojang.brigadier.tree.RootCommandNode;
 import com.mojang.logging.LogUtils;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Optional;
-import java.util.Set;
-import java.util.function.Consumer;
-import java.util.function.Predicate;
-import java.util.function.UnaryOperator;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 import net.minecraft.ChatFormatting;
 import net.minecraft.SharedConstants;
 import net.minecraft.commands.execution.ExecutionContext;
@@ -35,113 +26,14 @@ import net.minecraft.core.HolderSet;
 import net.minecraft.core.Registry;
 import net.minecraft.data.registries.VanillaRegistries;
 import net.minecraft.gametest.framework.TestCommand;
-import net.minecraft.network.chat.ClickEvent;
-import net.minecraft.network.chat.CommonComponents;
-import net.minecraft.network.chat.Component;
-import net.minecraft.network.chat.ComponentUtils;
-import net.minecraft.network.chat.HoverEvent;
-import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.*;
 import net.minecraft.network.protocol.game.ClientboundCommandsPacket;
 import net.minecraft.resources.Identifier;
 import net.minecraft.resources.ResourceKey;
-import net.minecraft.server.commands.AdvancementCommands;
-import net.minecraft.server.commands.AttributeCommand;
-import net.minecraft.server.commands.BanIpCommands;
-import net.minecraft.server.commands.BanListCommands;
-import net.minecraft.server.commands.BanPlayerCommands;
-import net.minecraft.server.commands.BossBarCommands;
-import net.minecraft.server.commands.ChaseCommand;
-import net.minecraft.server.commands.ClearInventoryCommands;
-import net.minecraft.server.commands.CloneCommands;
-import net.minecraft.server.commands.DamageCommand;
-import net.minecraft.server.commands.DataPackCommand;
-import net.minecraft.server.commands.DeOpCommands;
-import net.minecraft.server.commands.DebugCommand;
-import net.minecraft.server.commands.DebugConfigCommand;
-import net.minecraft.server.commands.DebugMobSpawningCommand;
-import net.minecraft.server.commands.DebugPathCommand;
-import net.minecraft.server.commands.DefaultGameModeCommands;
-import net.minecraft.server.commands.DialogCommand;
-import net.minecraft.server.commands.DifficultyCommand;
-import net.minecraft.server.commands.EffectCommands;
-import net.minecraft.server.commands.EmoteCommands;
-import net.minecraft.server.commands.EnchantCommand;
-import net.minecraft.server.commands.ExecuteCommand;
-import net.minecraft.server.commands.ExperienceCommand;
-import net.minecraft.server.commands.FetchProfileCommand;
-import net.minecraft.server.commands.FillBiomeCommand;
-import net.minecraft.server.commands.FillCommand;
-import net.minecraft.server.commands.ForceLoadCommand;
-import net.minecraft.server.commands.FunctionCommand;
-import net.minecraft.server.commands.GameModeCommand;
-import net.minecraft.server.commands.GameRuleCommand;
-import net.minecraft.server.commands.GiveCommand;
-import net.minecraft.server.commands.HelpCommand;
-import net.minecraft.server.commands.ItemCommands;
-import net.minecraft.server.commands.JfrCommand;
-import net.minecraft.server.commands.KickCommand;
-import net.minecraft.server.commands.KillCommand;
-import net.minecraft.server.commands.ListPlayersCommand;
-import net.minecraft.server.commands.LocateCommand;
-import net.minecraft.server.commands.LootCommand;
-import net.minecraft.server.commands.MsgCommand;
-import net.minecraft.server.commands.OpCommand;
-import net.minecraft.server.commands.PardonCommand;
-import net.minecraft.server.commands.PardonIpCommand;
-import net.minecraft.server.commands.ParticleCommand;
-import net.minecraft.server.commands.PerfCommand;
-import net.minecraft.server.commands.PlaceCommand;
-import net.minecraft.server.commands.PlaySoundCommand;
-import net.minecraft.server.commands.PublishCommand;
-import net.minecraft.server.commands.RaidCommand;
-import net.minecraft.server.commands.RandomCommand;
-import net.minecraft.server.commands.RecipeCommand;
-import net.minecraft.server.commands.ReloadCommand;
-import net.minecraft.server.commands.ReturnCommand;
-import net.minecraft.server.commands.RideCommand;
-import net.minecraft.server.commands.RotateCommand;
-import net.minecraft.server.commands.SaveAllCommand;
-import net.minecraft.server.commands.SaveOffCommand;
-import net.minecraft.server.commands.SaveOnCommand;
-import net.minecraft.server.commands.SayCommand;
-import net.minecraft.server.commands.ScheduleCommand;
-import net.minecraft.server.commands.ScoreboardCommand;
-import net.minecraft.server.commands.SeedCommand;
-import net.minecraft.server.commands.ServerPackCommand;
-import net.minecraft.server.commands.SetBlockCommand;
-import net.minecraft.server.commands.SetPlayerIdleTimeoutCommand;
-import net.minecraft.server.commands.SetSpawnCommand;
-import net.minecraft.server.commands.SetWorldSpawnCommand;
-import net.minecraft.server.commands.SpawnArmorTrimsCommand;
-import net.minecraft.server.commands.SpectateCommand;
-import net.minecraft.server.commands.SpreadPlayersCommand;
-import net.minecraft.server.commands.StopCommand;
-import net.minecraft.server.commands.StopSoundCommand;
-import net.minecraft.server.commands.StopwatchCommand;
-import net.minecraft.server.commands.SummonCommand;
-import net.minecraft.server.commands.TagCommand;
-import net.minecraft.server.commands.TeamCommand;
-import net.minecraft.server.commands.TeamMsgCommand;
-import net.minecraft.server.commands.TeleportCommand;
-import net.minecraft.server.commands.TellRawCommand;
-import net.minecraft.server.commands.TickCommand;
-import net.minecraft.server.commands.TimeCommand;
-import net.minecraft.server.commands.TitleCommand;
-import net.minecraft.server.commands.TransferCommand;
-import net.minecraft.server.commands.TriggerCommand;
-import net.minecraft.server.commands.VersionCommand;
-import net.minecraft.server.commands.WardenSpawnTrackerCommand;
-import net.minecraft.server.commands.WaypointCommand;
-import net.minecraft.server.commands.WeatherCommand;
-import net.minecraft.server.commands.WhitelistCommand;
-import net.minecraft.server.commands.WorldBorderCommand;
+import net.minecraft.server.commands.*;
 import net.minecraft.server.commands.data.DataCommands;
 import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.server.permissions.PermissionCheck;
-import net.minecraft.server.permissions.PermissionProviderCheck;
-import net.minecraft.server.permissions.PermissionSet;
-import net.minecraft.server.permissions.PermissionSetSupplier;
-import net.minecraft.server.permissions.Permissions;
+import net.minecraft.server.permissions.*;
 import net.minecraft.tags.TagKey;
 import net.minecraft.util.Util;
 import net.minecraft.util.profiling.Profiler;
@@ -154,6 +46,16 @@ import net.minecraft.world.phys.Vec3;
 import org.jspecify.annotations.Nullable;
 import org.slf4j.Logger;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
 public class Commands {
     public interface RestrictedMarker { } // Paper - restricted api
     public static final String COMMAND_PREFIX = "/";
@@ -297,7 +199,6 @@ public class Commands {
             StopCommand.register(this.dispatcher);
             TransferCommand.register(this.dispatcher);
             WhitelistCommand.register(this.dispatcher);
-            if (org.dreeam.leaf.config.modules.gameplay.AfkCommand.enabled) org.purpurmc.purpur.command.AFKCommand.register(this.dispatcher); // Leaf - Improve Purpur AFK system
             org.purpurmc.purpur.command.UptimeCommand.register(this.dispatcher); // Purpur - Add uptime command
             org.purpurmc.purpur.command.TPSBarCommand.register(this.dispatcher); // Purpur - Implement TPSBar
             org.purpurmc.purpur.command.RamBarCommand.register(this.dispatcher); // Purpur - Add rambar command
diff --git a/net/minecraft/gametest/framework/GameTestHelper.java b/net/minecraft/gametest/framework/GameTestHelper.java
index 49d3154afe2ca5789e63bdf972c791969405c16d..bdef46338e8d8eaebd77296f2ebac6a7ab6ae03e 100644
--- a/net/minecraft/gametest/framework/GameTestHelper.java
+++ b/net/minecraft/gametest/framework/GameTestHelper.java
@@ -322,8 +322,6 @@ public class GameTestHelper {
                 return gameType;
             }
 
-            public void setAfk(final boolean afk) {} // Purpur - AFK API
-
             public void resetLastActionTime() {} // Purpur - Ridables
 
             @Override
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 0e77c717d3481aae5d591a08eed1eb5f1439cdbf..236e32bfb6cffc95368037f3904a2e6efe698a9a 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -286,7 +286,6 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         thread.start(); // Paper - Enhance console tab completions for brigadier commands; start console thread after MinecraftServer.console & PaperConfig are initialized
         io.papermc.paper.command.PaperCommands.registerCommands(this); // Paper - setup /paper command
         this.server.spark.registerCommandBeforePlugins(this.server); // Paper - spark
-        com.destroystokyo.paper.Metrics.PaperMetrics.startMetrics(); // Paper - start metrics
         /*// Purpur start - Purpur config files // Purpur start - Migrate Setting to reintroduce end void rings
         try {
             org.purpurmc.purpur.PurpurConfig.init((java.io.File) options.valueOf("purpur-settings"));
diff --git a/net/minecraft/server/level/ServerPlayer.java b/net/minecraft/server/level/ServerPlayer.java
index c9db0f183e5607fde9cdf248b88fc357d2238d41..bd53374ecf90528a20b6a0bd41ab9c7e352a716f 100644
--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -2604,81 +2604,11 @@ public class ServerPlayer extends Player implements ca.spottedleaf.moonrise.patc
 
     public void resetLastActionTime() {
         this.lastActionTime = Util.getMillis();
-        this.setAfk(false); // Purpur - AFK API
-    }
-
-    // Purpur start - AFK API
-    private boolean isAfk = false;
-    // Leaf sart - Improve Purpur AFK system
-    public boolean isCommandAfk = false;
-    public boolean commandAfkStatus = false;
-    // Leaf end - Improve Purpur AFK system
-
-    @Override
-    public void setAfk(boolean afk) {
-        if (this.isAfk == afk) {
-            return;
-        }
-
-        String msg = afk ? org.purpurmc.purpur.PurpurConfig.afkBroadcastAway : org.purpurmc.purpur.PurpurConfig.afkBroadcastBack;
-
-        org.purpurmc.purpur.event.PlayerAFKEvent event = new org.purpurmc.purpur.event.PlayerAFKEvent(this.getBukkitEntity(), afk, this.level().purpurConfig.idleTimeoutKick, msg, !org.bukkit.Bukkit.isPrimaryThread());
-        if (!event.callEvent() || event.shouldKick()) {
-            return;
-        }
-
-        this.isAfk = afk;
-
-        if (!afk) {
-            resetLastActionTime();
-        }
-
-        msg = event.getBroadcastMsg();
-        if (msg != null && !msg.isEmpty()) {
-            String playerName = this.getGameProfile().name();
-            if (org.purpurmc.purpur.PurpurConfig.afkBroadcastUseDisplayName) {
-                net.kyori.adventure.text.Component playerDisplayNameComponent = net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(this.getBukkitEntity().getDisplayName());
-                playerName = net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer.plainText().serialize(playerDisplayNameComponent);
-            }
-            server.getPlayerList().broadcastMiniMessage(String.format(msg, playerName), false);
-        }
-
-        if (this.level().purpurConfig.idleTimeoutUpdateTabList) {
-            String scoreboardName = getScoreboardName();
-            String playerListName = net.kyori.adventure.text.minimessage.MiniMessage.miniMessage().serialize(getBukkitEntity().playerListName());
-            String[] split = playerListName.split(java.util.regex.Pattern.quote(scoreboardName)); // Leaf - Configurable vanilla username check - Sanitize name input
-            String prefix = (split.length > 0 ? split[0] : "").replace(org.purpurmc.purpur.PurpurConfig.afkTabListPrefix, "");
-            String suffix = (split.length > 1 ? split[1] : "").replace(org.purpurmc.purpur.PurpurConfig.afkTabListSuffix, "");
-            if (afk) {
-                // Leaf start - Improve Purpur AFK system
-                String[] rawTitle = org.purpurmc.purpur.PurpurConfig.afkTitleAway.split(":");
-                String title = rawTitle[0];
-                String subTitle = rawTitle[1];
-                long fadeInTicks = Long.parseLong(rawTitle[2]);
-                long stayTicks = Long.parseLong(rawTitle[3]);
-                long fadeOutTicks = Long.parseLong(rawTitle[4]);
-
-                net.kyori.adventure.title.Title tile = net.kyori.adventure.title.Title.title(net.kyori.adventure.text.minimessage.MiniMessage.miniMessage().deserialize(title), net.kyori.adventure.text.minimessage.MiniMessage.miniMessage().deserialize(subTitle), net.kyori.adventure.title.Title.Times.times(net.kyori.adventure.util.Ticks.duration(fadeInTicks), net.kyori.adventure.util.Ticks.duration(stayTicks), net.kyori.adventure.util.Ticks.duration(fadeOutTicks)));
-                getBukkitEntity().showTitle(tile);
-                // Leaf end - Improve Purpur AFK system
-
-                getBukkitEntity().setPlayerListName(org.purpurmc.purpur.PurpurConfig.afkTabListPrefix + prefix + scoreboardName + suffix + org.purpurmc.purpur.PurpurConfig.afkTabListSuffix, true);
-            } else {
-                getBukkitEntity().setPlayerListName(prefix + scoreboardName + suffix, true);
-            }
-        }
-
-        ((ServerLevel) this.level()).updateSleepingPlayerList();
-    }
-
-    @Override
-    public boolean isAfk() {
-        return this.isAfk;
     }
 
     @Override
     public boolean canBeCollidedWith(Entity entity) {
-        return !this.isAfk() && super.canBeCollidedWith(entity);
+        return super.canBeCollidedWith(entity);
     }
     // Purpur end - AFK API
 
diff --git a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
index 0e1411d820caef8a6a80bca07d9c76e9615ff13e..0ebdd65ce246331f9d27d58b193333e4b2a8ddd0 100644
--- a/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerCommonPacketListenerImpl.java
@@ -58,10 +58,6 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
     private static final long KEEPALIVE_LIMIT = KEEPALIVE_LIMIT_IN_SECONDS * 1000;
     // Gale end - Purpur - send multiple keep-alive packets
     protected static final net.minecraft.resources.Identifier MINECRAFT_BRAND = net.minecraft.resources.Identifier.withDefaultNamespace("brand"); // Paper - Brand support
-    // Purpur start - Purpur client support
-    protected static final net.minecraft.resources.Identifier PURPUR_CLIENT = net.minecraft.resources.Identifier.fromNamespaceAndPath("purpur", "client");
-    public boolean purpurClient;
-    // Purpur end - Purpur client support
     // Paper start - retain certain values
     public @Nullable String playerBrand;
     public final java.util.Set<String> pluginMessagerChannels;
@@ -206,12 +202,6 @@ public abstract class ServerCommonPacketListenerImpl implements ServerCommonPack
                 return;
             }
 
-            // Purpur start - Purpur client support
-            if (identifier.equals(PURPUR_CLIENT)) {
-                this.purpurClient = true;
-            }
-            // Purpur end - Purpur client support
-
             if (identifier.equals(MINECRAFT_BRAND)) {
                 this.playerBrand = new net.minecraft.network.FriendlyByteBuf(io.netty.buffer.Unpooled.wrappedBuffer(data)).readUtf(256);
             }
diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 2c824def972d5b9017090ecfcff816eed44e3676..814bb6ed843c708502783f253684fcab7e5be285 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -346,14 +346,12 @@ public class ServerGamePacketListenerImpl
     }
     // Paper end - configuration phase API
 
-    // Purpur start - AFK API
     // Leaf start - Use caffeine cache for kickPermission instead of using google.common.cache
     private final com.github.benmanes.caffeine.cache.LoadingCache<org.bukkit.craftbukkit.entity.CraftPlayer, Boolean> kickPermissionCache = com.github.benmanes.caffeine.cache.Caffeine.newBuilder()
         .maximumSize(1000)
         .expireAfterWrite(1, java.util.concurrent.TimeUnit.MINUTES)
         .build(player -> player.hasPermission("purpur.bypassIdleKick"));
     // Leaf end - Use caffeine cache for kickPermission instead of using google.common.cache
-    // Purpur end - AFK API
 
     @Override
     public void tick() {
@@ -373,12 +371,9 @@ public class ServerGamePacketListenerImpl
                 && this.server.playerIdleTimeout() > 0
                 && Util.getMillis() - this.player.getLastActionTime() > TimeUnit.MINUTES.toMillis(this.server.playerIdleTimeout())
                 && !this.player.wonGame) {
-                // Purpur start - AFK API
-                this.player.setAfk(true);
                 if (!this.player.level().purpurConfig.idleTimeoutKick || (!Boolean.parseBoolean(System.getenv("PURPUR_FORCE_IDLE_KICK")) && kickPermissionCache.get(this.player.getBukkitEntity()))) { // Leaf - Use caffeine cache for kickPermission instead of using google.common.cache
                     return;
                 }
-                // Purpur end - AFK API
                 this.disconnect(Component.translatable("multiplayer.disconnect.idling"), org.bukkit.event.player.PlayerKickEvent.Cause.IDLING); // Paper - kick event cause
             }
         }
@@ -698,8 +693,6 @@ public class ServerGamePacketListenerImpl
                     this.lastYaw = to.getYaw();
                     this.lastPitch = to.getPitch();
 
-                    if (!to.getWorld().getUID().equals(from.getWorld().getUID()) || to.getBlockX() != from.getBlockX() || to.getBlockY() != from.getBlockY() || to.getBlockZ() != from.getBlockZ() || to.getYaw() != from.getYaw() || to.getPitch() != from.getPitch()) this.player.resetLastActionTime(); // Purpur - AFK API
-
                     Location oldTo = to.clone();
                     PlayerMoveEvent event = new PlayerMoveEvent(player, from, to);
                     this.cserver.getPluginManager().callEvent(event);
@@ -1664,7 +1657,7 @@ public class ServerGamePacketListenerImpl
                                         movedWrongly = true;
                                         if (event.getLogWarning())
                                      // Paper end
-                                    LOGGER.warn("{} moved wrongly!, ({})", this.player.getPlainTextName(), verticalDelta); // Purpur - AFK API
+                                    LOGGER.warn("{} moved wrongly!, ({})", this.player.getName(), verticalDelta);
                                     } // Paper
                                 }
 
@@ -1729,8 +1722,6 @@ public class ServerGamePacketListenerImpl
                                         this.lastYaw = to.getYaw();
                                         this.lastPitch = to.getPitch();
 
-                                        if (!to.getWorld().getUID().equals(from.getWorld().getUID()) || to.getBlockX() != from.getBlockX() || to.getBlockY() != from.getBlockY() || to.getBlockZ() != from.getBlockZ() || to.getYaw() != from.getYaw() || to.getPitch() != from.getPitch()) this.player.resetLastActionTime(); // Purpur - AFK API
-
                                         Location oldTo = to.clone();
                                         PlayerMoveEvent event = new PlayerMoveEvent(player, from, to);
                                         this.cserver.getPluginManager().callEvent(event);
@@ -1786,13 +1777,6 @@ public class ServerGamePacketListenerImpl
                                         this.player.tryResetCurrentImpulseContext();
                                     }
 
-                                    // Purpur start - Dont run with scissors!
-                                    if (this.player.level().purpurConfig.dontRunWithScissors && this.player.isSprinting() && !(this.player.level().purpurConfig.ignoreScissorsInWater && this.player.isInWater()) && !(this.player.level().purpurConfig.ignoreScissorsInLava && this.player.isInLava()) && (isScissors(this.player.getItemInHand(InteractionHand.MAIN_HAND)) || isScissors(this.player.getItemInHand(InteractionHand.OFF_HAND))) && (int) (Math.random() * 10) == 0) {
-                                        this.player.hurtServer(this.player.level(), this.player.damageSources().scissors(), (float) this.player.level().purpurConfig.scissorsRunningDamage);
-                                        if (!org.purpurmc.purpur.PurpurConfig.dontRunWithScissors.isBlank()) this.player.sendActionBarMessage(org.purpurmc.purpur.PurpurConfig.dontRunWithScissors);
-                                    }
-                                    // Purpur end - Dont run with scissors!
-
                                     this.player.checkMovementStatistics(this.player.getX() - x, this.player.getY() - y, this.player.getZ() - z);
                                     this.lastGoodX = this.player.getX();
                                     this.lastGoodY = this.player.getY();
@@ -1810,17 +1794,6 @@ public class ServerGamePacketListenerImpl
         }
     }
 
-    // Purpur start - Dont run with scissors!
-    public boolean isScissors(ItemStack stack) {
-        if (!stack.is(Items.SHEARS)) return false;
-
-        Identifier itemModelReference = stack.get(net.minecraft.core.component.DataComponents.ITEM_MODEL);
-        if (itemModelReference != null && itemModelReference.equals(this.player.level().purpurConfig.dontRunWithScissorsItemModelReference)) return true;
-
-        return stack.getOrDefault(DataComponents.CUSTOM_MODEL_DATA, net.minecraft.world.item.component.CustomModelData.EMPTY).equals(net.minecraft.world.item.component.CustomModelData.EMPTY);
-    }
-    // Purpur end - Dont run with scissors!
-
     private boolean shouldCheckPlayerMovement(boolean isElytraMovement) {
         if (this.isSingleplayerOwner()) {
             return false;
@@ -2425,7 +2398,6 @@ public class ServerGamePacketListenerImpl
 
     @Override
     public void handleChatCommand(ServerboundChatCommandPacket packet) {
-        final boolean isAfkCommandCooldown = this.performAfkCooldown(packet.command()); // Leaf - Improve Purpur AFK system
         this.tryHandleChat(packet.command(), true, () -> {
             // CraftBukkit start - SPIGOT-7346: Prevent disconnected players from executing commands
             if (this.player.hasDisconnected() || this.server.isStopped()) { // Leaf - Prevent executing commands if server stopped
@@ -2434,7 +2406,7 @@ public class ServerGamePacketListenerImpl
             // CraftBukkit end
             this.performUnsignedChatCommand(packet.command());
             this.detectRateSpam("/" + packet.command()); // Spigot
-        }, true, isAfkCommandCooldown); // CraftBukkit - sync commands // Leaf - Improve Purpur AFK system
+        }, true); // CraftBukkit - sync commands
     }
 
     private void performUnsignedChatCommand(String command) {
@@ -2467,7 +2439,6 @@ public class ServerGamePacketListenerImpl
     public void handleSignedChatCommand(ServerboundChatCommandSignedPacket packet) {
         Optional<LastSeenMessages> optional = this.unpackAndApplyLastSeen(packet.lastSeenMessages());
         if (!optional.isEmpty()) {
-            final boolean isAfkCommandCooldown = this.performAfkCooldown(packet.command()); // Leaf - Improve Purpur AFK system
             this.tryHandleChat(packet.command(), true, () -> {
                 // CraftBukkit start - SPIGOT-7346: Prevent disconnected players from executing commands
                 if (this.player.hasDisconnected() || this.server.isStopped()) { // Leaf - Prevent executing commands if server stopped
@@ -2476,7 +2447,7 @@ public class ServerGamePacketListenerImpl
                 // CraftBukkit end
                 this.performSignedChatCommand(packet, optional.get());
                 this.detectRateSpam("/" + packet.command()); // Spigot
-            }, true, isAfkCommandCooldown); // CraftBukkit - sync commands // Leaf - Improve Purpur AFK system
+            }, true); // CraftBukkit - sync commands
         }
     }
 
@@ -2583,17 +2554,12 @@ public class ServerGamePacketListenerImpl
         return dispatcher.parse(command, this.player.createCommandSourceStack());
     }
 
-    private void tryHandleChat(String message, boolean bypassHiddenChat, Runnable handler, boolean sync) { // CraftBukkit
-        // Leaf start - Improve Purpur AFK system
-        tryHandleChat(message, bypassHiddenChat, handler, sync, false);
-    }
-    private void tryHandleChat(String message, boolean bypassHiddenChat, Runnable handler, boolean sync, boolean afkCommandCooldown) {
-        // Leaf end - Improve Purpur AFK system
+    private void tryHandleChat(String message, boolean bypassHiddenChat, Runnable handler, boolean sync) {
         if (isChatMessageIllegal(message)) {
             this.disconnectAsync(Component.translatable("multiplayer.disconnect.illegal_characters"), org.bukkit.event.player.PlayerKickEvent.Cause.ILLEGAL_CHARACTERS); // Paper - add proper async disconnect
         } else if (this.player.isRemoved() || (!bypassHiddenChat && this.player.getChatVisibility() == ChatVisiblity.HIDDEN)) { // CraftBukkit - dead men tell no tales
             this.send(new ClientboundSystemChatPacket(Component.translatable("chat.disabled.options").withStyle(ChatFormatting.RED), false));
-        } else if (player.didPlayerJoinEvent && !afkCommandCooldown) { // Gale - EMC - do not process chat/commands before player has joined // Leaf - Improve Purpur AFK system - don't process afk command if in cooldown
+        } else if (player.didPlayerJoinEvent) { // Gale - EMC - do not process chat/commands before player has joined
             this.player.resetLastActionTime();
             // CraftBukkit start
             if (sync) {
@@ -2605,40 +2571,6 @@ public class ServerGamePacketListenerImpl
         }
     }
 
-    // Leaf start - Improve Purpur AFK system
-    public static final Map<java.util.UUID, Long> afkCooldown = new it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap<>();
-    private boolean performAfkCooldown(String command) {
-        if (!org.dreeam.leaf.config.modules.gameplay.AfkCommand.enabled || !"afk".equals(command)) {
-            return false;
-        }
-
-        this.player.commandAfkStatus = this.player.isAfk();
-        this.player.isCommandAfk = true;
-
-        if (org.purpurmc.purpur.PurpurConfig.afkCommandCooldown <= 0) {
-            return false;
-        }
-
-        java.util.UUID uuid = this.player.getUUID();
-        Long cooldown = afkCooldown.get(uuid);
-        long currentTime = System.nanoTime();
-
-        if (cooldown != null && (currentTime - cooldown) / 1_000_000_000 <= org.purpurmc.purpur.PurpurConfig.afkCommandCooldown) {
-            String msg = org.purpurmc.purpur.PurpurConfig.afkCooldown;
-
-            if (msg != null && !msg.isEmpty()) {
-                net.kyori.adventure.text.Component message = net.kyori.adventure.text.minimessage.MiniMessage.miniMessage().deserialize(msg.replace("%time%", String.valueOf(org.purpurmc.purpur.PurpurConfig.afkCommandCooldown - (currentTime - cooldown) / 1_000_000_000)));
-                this.player.sendMessage(message);
-            }
-
-            return true;
-        } else {
-            afkCooldown.put(uuid, currentTime);
-            return false;
-        }
-    }
-    // Leaf end - Improve Purpur AFK system
-
     private Optional<LastSeenMessages> unpackAndApplyLastSeen(LastSeenMessages.Update update) {
         synchronized (this.lastSeenMessages) {
             Optional var10000;
diff --git a/net/minecraft/server/players/PlayerList.java b/net/minecraft/server/players/PlayerList.java
index a6ab772731afa1b339c16fc91aa915cd35c71f4d..1d20f58c66750f97dcb0d19a337518f82982e19e 100644
--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -458,7 +458,6 @@ public abstract class PlayerList {
     public net.kyori.adventure.text.@Nullable Component remove(ServerPlayer player, net.kyori.adventure.text.Component leaveMessage) {
         // Paper end - Fix kick event leave message not being sent
         org.purpurmc.purpur.task.BossBarTask.removeFromAll(player.getBukkitEntity()); // Purpur - Implement TPSBar
-        net.minecraft.server.network.ServerGamePacketListenerImpl.afkCooldown.remove(player.getBukkitEntity().getUniqueId()); // Leaf - Improve Purpur AFK system
         ServerLevel serverLevel = player.level();
         player.awardStat(Stats.LEAVE_GAME);
         // CraftBukkit start - Quitting must be before we do final save of data, in case plugins need to modify it
diff --git a/net/minecraft/server/players/SleepStatus.java b/net/minecraft/server/players/SleepStatus.java
index e6827e90b685f88d945010f2c8c5aead52b0856e..a0b9e3143bcad1f443c3935dca7457b63ab523dd 100644
--- a/net/minecraft/server/players/SleepStatus.java
+++ b/net/minecraft/server/players/SleepStatus.java
@@ -26,7 +26,7 @@ public class SleepStatus {
                 anyPlayerSleeping = true;
             }
 
-            if (isSleepingLongEnough || player.fauxSleeping || (player.level().purpurConfig.idleTimeoutCountAsSleeping && player.isAfk())) { // Purpur - AFK API
+            if (isSleepingLongEnough || player.fauxSleeping) {
                 count++;
             }
         }
@@ -58,7 +58,7 @@ public class SleepStatus {
         for (ServerPlayer serverPlayer : players) {
             if (!serverPlayer.isSpectator()) {
                 this.activePlayers++;
-                if (serverPlayer.isSleeping() || serverPlayer.fauxSleeping || (serverPlayer.level().purpurConfig.idleTimeoutCountAsSleeping && serverPlayer.isAfk())) { // CraftBukkit // Purpur - AFK API
+                if (serverPlayer.isSleeping() || serverPlayer.fauxSleeping) { // CraftBukkit // Purpur - AFK API
                     this.sleepingPlayers++;
                 }
                 // CraftBukkit start
diff --git a/net/minecraft/world/damagesource/CombatTracker.java b/net/minecraft/world/damagesource/CombatTracker.java
index 30d0807d7adf46a6cd23b81b979e06f943a2bca1..6daa1c276f06dd67f9f099e17b75a3b8f07c6f01 100644
--- a/net/minecraft/world/damagesource/CombatTracker.java
+++ b/net/minecraft/world/damagesource/CombatTracker.java
@@ -107,15 +107,6 @@ public class CombatTracker {
                 Component component = ComponentUtils.wrapInSquareBrackets(Component.translatable(string + ".link")).withStyle(INTENTIONAL_GAME_DESIGN_STYLE);
                 return Component.translatable(string + ".message", this.mob.getDisplayName(), component);
             } else {
-                // Purpur start - Dont run with scissors!
-                if (damageSource.isScissors()) {
-                    return damageSource.getLocalizedDeathMessage(org.purpurmc.purpur.PurpurConfig.deathMsgRunWithScissors, this.mob);
-                // Purpur start - Stonecutter damage
-                } else if (damageSource.isStonecutter()) {
-                    return damageSource.getLocalizedDeathMessage(org.purpurmc.purpur.PurpurConfig.deathMsgStonecutter, this.mob);
-                // Purpur end - Stonecutter damage
-                }
-                // Purpur end - Dont run with scissors!
                 return damageSource.getLocalizedDeathMessage(this.mob);
             }
         }
diff --git a/net/minecraft/world/damagesource/DamageSource.java b/net/minecraft/world/damagesource/DamageSource.java
index 629c8a97e12f10313bb4540cd9379e4fa3cbd544..229c0e891b89bb0083ed7333a6dd40ddce738b81 100644
--- a/net/minecraft/world/damagesource/DamageSource.java
+++ b/net/minecraft/world/damagesource/DamageSource.java
@@ -24,7 +24,6 @@ public class DamageSource {
     private org.bukkit.block.@Nullable Block eventBlockDamager; // Relevant block set. damageSourcePosition is only used for bad respawn point explosion or custom damage
     private org.bukkit.block.@Nullable BlockState fromBlockSnapshot; // Captured block snapshot when the eventBlockDamager is not relevant (e.g. for bad respawn point explosions the block is already removed)
     private boolean critical; // Supports arrows and sweeping damage
-    private boolean scissors = false; // Purpur - Dont run with scissors!
     private boolean stonecutter = false; // Purpur - Stonecutter damage
 
     public DamageSource knownCause(final org.bukkit.event.entity.EntityDamageEvent.DamageCause cause) {
@@ -37,30 +36,6 @@ public class DamageSource {
         return this.knownCause;
     }
 
-    // Purpur start - Dont run with scissors!
-    public DamageSource scissors() {
-        this.knownCause(org.bukkit.event.entity.EntityDamageEvent.DamageCause.SUICIDE);
-        this.scissors = true;
-        return this;
-    }
-
-    public boolean isScissors() {
-        return this.scissors;
-    }
-    // Purpur end - Dont run with scissors!
-
-    // Purpur start -  - Stonecutter damage
-    public DamageSource stonecutter() {
-        this.knownCause(org.bukkit.event.entity.EntityDamageEvent.DamageCause.CONTACT);
-        this.stonecutter = true;
-        return this;
-    }
-
-    public boolean isStonecutter() {
-        return this.stonecutter;
-    }
-    // Purpur end - Stonecutter damage
-
     @Nullable
     public Entity eventEntityDamager() {
         return this.eventEntityDamager;
@@ -120,8 +95,6 @@ public class DamageSource {
         damageSource.eventBlockDamager = this.eventBlockDamager;
         damageSource.fromBlockSnapshot = this.fromBlockSnapshot;
         damageSource.critical = this.critical;
-        damageSource.scissors = this.isScissors(); // Purpur - Dont run with scissors!
-        damageSource.stonecutter = this.isStonecutter(); // Purpur - Stonecutter damage
         return damageSource;
     }
     // CraftBukkit end
diff --git a/net/minecraft/world/damagesource/DamageSources.java b/net/minecraft/world/damagesource/DamageSources.java
index 1e1b773d88917b251d71aad40b85fbc5fffce8c8..e45891d10dfab44d3aa584cc7db3fbc3c7654af5 100644
--- a/net/minecraft/world/damagesource/DamageSources.java
+++ b/net/minecraft/world/damagesource/DamageSources.java
@@ -42,8 +42,6 @@ public class DamageSources {
     private final DamageSource stalagmite;
     private final DamageSource outsideBorder;
     private final DamageSource genericKill;
-    private final DamageSource scissors; // Purpur - Dont run with scissors!
-    private final DamageSource stonecutter; // Purpur - Stonecutter damage
 
     public DamageSources(RegistryAccess registry) {
         this.damageTypes = registry.lookupOrThrow(Registries.DAMAGE_TYPE);
@@ -72,8 +70,6 @@ public class DamageSources {
         this.stalagmite = this.source(DamageTypes.STALAGMITE);
         this.outsideBorder = this.source(DamageTypes.OUTSIDE_BORDER);
         this.genericKill = this.source(DamageTypes.GENERIC_KILL);
-        this.scissors = this.source(DamageTypes.MAGIC).scissors(); // Purpur - Dont run with scissors!
-        this.stonecutter = this.source(DamageTypes.MAGIC).stonecutter(); // Purpur - Stonecutter damage
     }
 
     private DamageSource source(ResourceKey<DamageType> damageTypeKey) {
@@ -88,18 +84,6 @@ public class DamageSources {
         return new DamageSource(this.damageTypes.getOrThrow(damageTypeKey), causingEntity, directEntity);
     }
 
-    // Purpur start - Dont run with scissor
-    public DamageSource scissors() {
-        return this.scissors;
-    }
-    // Purpur end - Dont run with scissors!
-
-    // Purpur start - Stonecutter damage
-    public DamageSource stonecutter() {
-        return this.stonecutter;
-    }
-    // Purpur end - Stonecutter damage
-
     public DamageSource inFire() {
         return this.inFire;
     }
diff --git a/net/minecraft/world/entity/EntitySelector.java b/net/minecraft/world/entity/EntitySelector.java
index a0d23f7c733c315a021e8b2ba8968f41807159a4..326da0817707fb2cd145c2c387d88b5e6ad780d4 100644
--- a/net/minecraft/world/entity/EntitySelector.java
+++ b/net/minecraft/world/entity/EntitySelector.java
@@ -27,8 +27,6 @@ public final class EntitySelector {
         net.minecraft.server.level.ServerPlayer serverPlayer = (net.minecraft.server.level.ServerPlayer) player;
         return net.minecraft.util.Mth.clamp(serverPlayer.getStats().getValue(net.minecraft.stats.Stats.CUSTOM.get(net.minecraft.stats.Stats.TIME_SINCE_REST)), 1, Integer.MAX_VALUE) >= playerInsomniaTicks;
     };
-    // Paper end - Ability to control player's insomnia and phantoms
-    public static Predicate<Player> notAfk = (player) -> !player.isAfk(); // Purpur - AFK API
 
     // Paper start - Affects Spawning API
     public static final Predicate<Entity> PLAYER_AFFECTS_SPAWNING = (entity) -> {
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 5d7076e4573d47ea30d63dd16f0788012bd39264..128d8d936f5b574003ba54fd6545716f66d874fb 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1511,24 +1511,6 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
                 this.stopSleeping();
             }
 
-            // Purpur start - One Punch Man!
-            if (damageSource.getEntity() instanceof net.minecraft.world.entity.player.Player player && damageSource.getEntity().level().purpurConfig.creativeOnePunch && !damageSource.is(DamageTypeTags.IS_PROJECTILE)) {
-                if (player.isCreative()) {
-                    org.apache.commons.lang3.mutable.MutableDouble attackDamage = new org.apache.commons.lang3.mutable.MutableDouble();
-                    player.getMainHandItem().forEachModifier(EquipmentSlot.MAINHAND, (attributeHolder, attributeModifier) -> {
-                        if (attributeModifier.operation() == AttributeModifier.Operation.ADD_VALUE) {
-                            attackDamage.addAndGet(attributeModifier.amount());
-                        }
-                    });
-
-                    if (attackDamage.doubleValue() == 0.0D) {
-                        // One punch!
-                        amount = 9999F;
-                    }
-                }
-            }
-            // Purpur end - One Punch Man!
-
             this.noActionTime = 0;
             if (amount < 0.0F) {
                 amount = 0.0F;
diff --git a/net/minecraft/world/entity/ai/targeting/TargetingConditions.java b/net/minecraft/world/entity/ai/targeting/TargetingConditions.java
index 0400376b2229811ee382c802222b05dd575d5633..b779c1550b9e0d44a314d8c304de2bcb19f6fcd9 100644
--- a/net/minecraft/world/entity/ai/targeting/TargetingConditions.java
+++ b/net/minecraft/world/entity/ai/targeting/TargetingConditions.java
@@ -63,10 +63,6 @@ public class TargetingConditions {
             return false;
         } else if (this.selector != null && !this.selector.test(target, level)) {
             return false;
-        // Purpur start - AFK API
-        } else if (!level.purpurConfig.idleTimeoutTargetPlayer && target instanceof net.minecraft.server.level.ServerPlayer player && player.isAfk()) {
-            return false;
-        // Purpur end - AFK API
         } else {
             if (entity == null) {
                 if (this.isCombat && (!target.canBeSeenAsEnemy() || level.getDifficulty() == Difficulty.PEACEFUL)) {
diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index e56fd037860c4f38862e01f7ab9972a7d7a132b2..8a4a0c6f139df3b7d5f598f4b8045101297a54f0 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -187,13 +187,6 @@ public abstract class Player extends Avatar implements ContainerUser {
     public boolean fauxSleeping;
     public int oldLevel = -1;
 
-    // Purpur start - AFK API
-    public abstract void setAfk(boolean afk);
-
-    public boolean isAfk() {
-        return false;
-    }
-    // Purpur end - AFK API
     @Override
     public org.bukkit.craftbukkit.entity.CraftHumanEntity getBukkitEntity() {
         return (org.bukkit.craftbukkit.entity.CraftHumanEntity) super.getBukkitEntity();
diff --git a/net/minecraft/world/entity/projectile/throwableitemprojectile/Snowball.java b/net/minecraft/world/entity/projectile/throwableitemprojectile/Snowball.java
index 329af7b66500519fdb0a938f0e7ff5bd4354f66d..41a893cf6aaef7c77cf1c73bc5ace81e2478c57d 100644
--- a/net/minecraft/world/entity/projectile/throwableitemprojectile/Snowball.java
+++ b/net/minecraft/world/entity/projectile/throwableitemprojectile/Snowball.java
@@ -73,22 +73,6 @@ public class Snowball extends ThrowableItemProjectile {
             net.minecraft.core.BlockPos relativePos = pos.relative(blockHitResult.getDirection());
 
             net.minecraft.world.level.block.state.BlockState blockState = this.level().getBlockState(pos);
-
-            if (this.level().purpurConfig.snowballExtinguishesFire && this.level().getBlockState(relativePos).is(net.minecraft.world.level.block.Blocks.FIRE)) {
-                if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(this, relativePos, net.minecraft.world.level.block.Blocks.AIR.defaultBlockState())) {
-                    this.level().removeBlock(relativePos, false);
-                }
-            } else if (this.level().purpurConfig.snowballExtinguishesCandles && net.minecraft.world.level.block.AbstractCandleBlock.isLit(blockState)) {
-                if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(this, pos, blockState.setValue(net.minecraft.world.level.block.AbstractCandleBlock.LIT, false))) {
-                    net.minecraft.world.level.block.AbstractCandleBlock.extinguish(null, blockState, this.level(), pos);
-                }
-            } else if (this.level().purpurConfig.snowballExtinguishesCampfires && net.minecraft.world.level.block.CampfireBlock.isLitCampfire(blockState)) {
-                if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(this, pos, blockState.setValue(net.minecraft.world.level.block.CampfireBlock.LIT, false))) {
-                    this.level().levelEvent(null, 1009, pos, 0);
-                    net.minecraft.world.level.block.CampfireBlock.dowse(this.getOwner(), this.level(), pos, blockState);
-                    this.level().setBlockAndUpdate(pos, blockState.setValue(net.minecraft.world.level.block.CampfireBlock.LIT, false));
-                }
-            }
         }
     }
     // Purpur end - options to extinguish fire blocks with snowballs
diff --git a/net/minecraft/world/level/EntityGetter.java b/net/minecraft/world/level/EntityGetter.java
index 6e91a8a045c0174363afb82c4cc7fce05cc39170..05f32f0fbc91d3251e53edc41410eb1bb353b734 100644
--- a/net/minecraft/world/level/EntityGetter.java
+++ b/net/minecraft/world/level/EntityGetter.java
@@ -289,7 +289,7 @@ public interface EntityGetter extends ca.spottedleaf.moonrise.patches.chunk_syst
 
     default boolean hasNearbyAlivePlayer(double x, double y, double z, double distance) {
         for (Player player : this.players()) {
-            if (EntitySelector.NO_SPECTATORS.test(player) && EntitySelector.LIVING_ENTITY_STILL_ALIVE.test(player) && EntitySelector.notAfk.test(player)) { // Purpur - AFK API
+            if (EntitySelector.NO_SPECTATORS.test(player) && EntitySelector.LIVING_ENTITY_STILL_ALIVE.test(player)) {
                 double d = player.distanceToSqr(x, y, z);
                 if (distance < 0.0 || d < distance * distance) {
                     return true;
diff --git a/net/minecraft/world/level/block/StonecutterBlock.java b/net/minecraft/world/level/block/StonecutterBlock.java
index 6ab51fcf42b9ca7ab46ea66498eea14250fc7444..835e071a346fa720e6ab80f93a3a60fd36cdc05c 100644
--- a/net/minecraft/world/level/block/StonecutterBlock.java
+++ b/net/minecraft/world/level/block/StonecutterBlock.java
@@ -93,13 +93,4 @@ public class StonecutterBlock extends Block {
         return false;
     }
 
-    // Purpur start - Stonecutter damage
-    @Override
-    public void stepOn(Level level, BlockPos pos, BlockState state, net.minecraft.world.entity.Entity entity) {
-        if (level.purpurConfig.stonecutterDamage > 0.0F && entity instanceof net.minecraft.world.entity.LivingEntity) {
-            entity.hurtServer((net.minecraft.server.level.ServerLevel) level, entity.damageSources().stonecutter().eventBlockDamager(level, pos), level.purpurConfig.stonecutterDamage);
-        }
-        super.stepOn(level, pos, state, entity);
-    }
-    // Purpur end - Stonecutter damage
 }
diff --git a/org/purpurmc/purpur/PurpurConfig.java b/org/purpurmc/purpur/PurpurConfig.java
index 4a8ac0345867d3a64426c4aec7b65770ea494c96..878d608f4415cc10d2ea6c3b1af4298311683907 100644
--- a/org/purpurmc/purpur/PurpurConfig.java
+++ b/org/purpurmc/purpur/PurpurConfig.java
@@ -173,20 +173,9 @@ public class PurpurConfig {
     }
 
     public static String cannotRideMob = "<red>You cannot mount that mob";
-    public static String afkBroadcastAway = "<yellow><italic>%s is now AFK";
-    public static String afkBroadcastBack = "<yellow><italic>%s is no longer AFK";
-    // Leaf start - Improve Purpur AFK system
-    public static String afkTitleAway = "<gold><bold>AFK:<red>You are now AFK...:10:70:20";
-    public static int afkCommandCooldown = 0;
-    public static String afkCooldown = "<gray>You need to wait %time%s to use /afk.";
-    // Leaf end - Improve Purpur AFK system
-    public static boolean afkBroadcastUseDisplayName = false;
-    public static String afkTabListPrefix = "[AFK] ";
-    public static String afkTabListSuffix = "";
     public static String ramCommandOutput = "<green>Ram Usage: <used>/<xmx> (<percent>)";
     public static String rambarCommandOutput = "<green>Rambar toggled <onoff> for <target>";
     public static String tpsbarCommandOutput = "<green>Tpsbar toggled <onoff> for <target>";
-    public static String dontRunWithScissors = "<red><italic>Don't run with scissors!";
     public static String uptimeCommandOutput = "<green>Server uptime is <uptime>";
     public static String unverifiedUsername = "default";
     public static String sleepSkippingNight = "default";
@@ -194,24 +183,9 @@ public class PurpurConfig {
     public static String sleepNotPossible = "default";
     private static void messages() {
         cannotRideMob = getString("settings.messages.cannot-ride-mob", cannotRideMob);
-        afkBroadcastAway = getString("settings.messages.afk-broadcast-away", afkBroadcastAway);
-        afkBroadcastBack = getString("settings.messages.afk-broadcast-back", afkBroadcastBack);
-        // Leaf start - Improve Purpur AFK system
-        afkTitleAway = getString("settings.messages.afk-title-away", afkTitleAway);
-        if (afkTitleAway.split(":").length != 5) {
-            Bukkit.getLogger().log(Level.SEVERE, "You put wrong format of afk-title-away in PurpurConfig, it should look like <gold><bold>AFK:<red>You are now AFK...:10:70:20");
-            Bukkit.getLogger().log(Level.SEVERE, "'Title:Sub Title:Title Fade In Ticks:Title Stay Ticks:Title Fade Out Ticks', split with :");
-        }
-        afkCommandCooldown = getInt("settings.messages.afk-command-cooldown", afkCommandCooldown);
-        afkCooldown = MiniMessage.miniMessage().serialize(MiniMessage.miniMessage().deserialize(getString("settings.messages.afk-command-cooldown-msg", afkCooldown)));
-        // Leaf end - Improve Purpur AFK system
-        afkBroadcastUseDisplayName = getBoolean("settings.messages.afk-broadcast-use-display-name", afkBroadcastUseDisplayName);
-        afkTabListPrefix = MiniMessage.miniMessage().serialize(MiniMessage.miniMessage().deserialize(getString("settings.messages.afk-tab-list-prefix", afkTabListPrefix)));
-        afkTabListSuffix = MiniMessage.miniMessage().serialize(MiniMessage.miniMessage().deserialize(getString("settings.messages.afk-tab-list-suffix", afkTabListSuffix)));
         ramCommandOutput = getString("settings.messages.ram-command-output", ramCommandOutput);
         rambarCommandOutput = getString("settings.messages.rambar-command-output", rambarCommandOutput);
         tpsbarCommandOutput = getString("settings.messages.tpsbar-command-output", tpsbarCommandOutput);
-        dontRunWithScissors = getString("settings.messages.dont-run-with-scissors", dontRunWithScissors);
         uptimeCommandOutput = getString("settings.messages.uptime-command-output", uptimeCommandOutput);
         unverifiedUsername = getString("settings.messages.unverified-username", unverifiedUsername);
         sleepSkippingNight = getString("settings.messages.sleep-skipping-night", sleepSkippingNight);
@@ -219,13 +193,6 @@ public class PurpurConfig {
         sleepNotPossible = getString("settings.messages.sleep-not-possible", sleepNotPossible);
     }
 
-    public static String deathMsgRunWithScissors = "<player> slipped and fell on their shears";
-    public static String deathMsgStonecutter = "<player> has sawed themself in half";
-    private static void deathMessages() {
-        deathMsgRunWithScissors = getString("settings.messages.death-message.run-with-scissors", deathMsgRunWithScissors);
-        deathMsgStonecutter = getString("settings.messages.death-message.stonecutter", deathMsgStonecutter);
-    }
-
     public static boolean advancementOnlyBroadcastToAffectedPlayer = false;
     public static boolean deathMessageOnlyBroadcastToAffectedPlayer = false;
     private static void broadcastSettings() {
diff --git a/org/purpurmc/purpur/PurpurWorldConfig.java b/org/purpurmc/purpur/PurpurWorldConfig.java
index 59d54268da0c053cdaec541076f5551d7b58cb65..388e84cbf8a3c8056e8611d9fc9327911da30d83 100644
--- a/org/purpurmc/purpur/PurpurWorldConfig.java
+++ b/org/purpurmc/purpur/PurpurWorldConfig.java
@@ -275,20 +275,12 @@ public class PurpurWorldConfig {
     public List<Item> itemImmuneToExplosion = new ArrayList<>();
     public List<Item> itemImmuneToFire = new ArrayList<>();
     public List<Item> itemImmuneToLightning = new ArrayList<>();
-    public boolean dontRunWithScissors = false;
-    public Identifier dontRunWithScissorsItemModelReference = Identifier.parse("purpurmc:scissors");
-    public boolean ignoreScissorsInWater = false;
-    public boolean ignoreScissorsInLava = false;
-    public double scissorsRunningDamage = 1D;
     public float enderPearlDamage = 5.0F;
     public int enderPearlCooldown = 20;
     public int enderPearlCooldownCreative = 20;
     public float enderPearlEndermiteChance = 0.05F;
     public int glowBerriesEatGlowDuration = 0;
     public boolean shulkerBoxItemDropContentsWhenDestroyed = true;
-    public boolean snowballExtinguishesFire = false;
-    public boolean snowballExtinguishesCandles = false;
-    public boolean snowballExtinguishesCampfires = false;
     private void itemSettings() {
         itemImmuneToCactus.clear();
         getList("gameplay-mechanics.item.immune.cactus", new ArrayList<>()).forEach(key -> {
@@ -326,20 +318,12 @@ public class PurpurWorldConfig {
             Item item = BuiltInRegistries.ITEM.getValue(Identifier.parse(key.toString()));
             if (item != Items.AIR) itemImmuneToLightning.add(item);
         });
-        dontRunWithScissors = getBoolean("gameplay-mechanics.item.shears.damage-if-sprinting", dontRunWithScissors);
-        dontRunWithScissorsItemModelReference = Identifier.parse(getString("gameplay-mechanics.item.shears.damage-if-sprinting-item-model", "purpurmc:scissors"));
-        ignoreScissorsInWater = getBoolean("gameplay-mechanics.item.shears.ignore-in-water", ignoreScissorsInWater);
-        ignoreScissorsInLava = getBoolean("gameplay-mechanics.item.shears.ignore-in-lava", ignoreScissorsInLava);
-        scissorsRunningDamage = getDouble("gameplay-mechanics.item.shears.sprinting-damage", scissorsRunningDamage);
         enderPearlDamage = (float) getDouble("gameplay-mechanics.item.ender-pearl.damage", enderPearlDamage);
         enderPearlCooldown = getInt("gameplay-mechanics.item.ender-pearl.cooldown", enderPearlCooldown);
         enderPearlCooldownCreative = getInt("gameplay-mechanics.item.ender-pearl.creative-cooldown", enderPearlCooldownCreative);
         enderPearlEndermiteChance = (float) getDouble("gameplay-mechanics.item.ender-pearl.endermite-spawn-chance", enderPearlEndermiteChance);
         glowBerriesEatGlowDuration = getInt("gameplay-mechanics.item.glow_berries.eat-glow-duration", glowBerriesEatGlowDuration);
         shulkerBoxItemDropContentsWhenDestroyed = getBoolean("gameplay-mechanics.item.shulker_box.drop-contents-when-destroyed", shulkerBoxItemDropContentsWhenDestroyed);
-        snowballExtinguishesFire = getBoolean("gameplay-mechanics.item.snowball.extinguish.fire", snowballExtinguishesFire);
-        snowballExtinguishesCandles = getBoolean("gameplay-mechanics.item.snowball.extinguish.candles", snowballExtinguishesCandles);
-        snowballExtinguishesCampfires = getBoolean("gameplay-mechanics.item.snowball.extinguish.campfires", snowballExtinguishesCampfires);
     }
 
     public double minecartMaxSpeed = 0.4D;
@@ -457,17 +441,12 @@ public class PurpurWorldConfig {
     }
 
     public boolean idleTimeoutKick = true;
-    public boolean idleTimeoutTickNearbyEntities = true;
-    public boolean idleTimeoutCountAsSleeping = false;
-    public boolean idleTimeoutUpdateTabList = false;
-    public boolean idleTimeoutTargetPlayer = true;
     public String playerDeathExpDropEquation = "expLevel * 7";
     public int playerDeathExpDropMax = 100;
     public boolean teleportIfOutsideBorder = false;
     public boolean teleportOnNetherCeilingDamage = false;
     public boolean totemOfUndyingWorksInInventory = false;
     public boolean playerFixStuckPortal = false;
-    public boolean creativeOnePunch = false;
     public boolean playerSleepNearMonsters = false;
     public boolean playersSkipNight = true;
     public double playerCriticalDamageMultiplier = 1.5D;
@@ -479,23 +458,13 @@ public class PurpurWorldConfig {
     public int playerExpPickupDelay = 2;
     public boolean playerVoidTrading = false;
     private void playerSettings() {
-        if (PurpurConfig.version < 19) {
-            boolean oldVal = getBoolean("gameplay-mechanics.player.idle-timeout.mods-target", idleTimeoutTargetPlayer);
-            set("gameplay-mechanics.player.idle-timeout.mods-target", null);
-            set("gameplay-mechanics.player.idle-timeout.mobs-target", oldVal);
-        }
         idleTimeoutKick = System.getenv("PURPUR_FORCE_IDLE_KICK") == null ? getBoolean("gameplay-mechanics.player.idle-timeout.kick-if-idle", idleTimeoutKick) : Boolean.parseBoolean(System.getenv("PURPUR_FORCE_IDLE_KICK"));
-        idleTimeoutTickNearbyEntities = getBoolean("gameplay-mechanics.player.idle-timeout.tick-nearby-entities", idleTimeoutTickNearbyEntities);
-        idleTimeoutCountAsSleeping = getBoolean("gameplay-mechanics.player.idle-timeout.count-as-sleeping", idleTimeoutCountAsSleeping);
-        idleTimeoutUpdateTabList = getBoolean("gameplay-mechanics.player.idle-timeout.update-tab-list", idleTimeoutUpdateTabList);
-        idleTimeoutTargetPlayer = getBoolean("gameplay-mechanics.player.idle-timeout.mobs-target", idleTimeoutTargetPlayer);
         playerDeathExpDropEquation = getString("gameplay-mechanics.player.exp-dropped-on-death.equation", playerDeathExpDropEquation);
         playerDeathExpDropMax = getInt("gameplay-mechanics.player.exp-dropped-on-death.maximum", playerDeathExpDropMax);
         teleportIfOutsideBorder = getBoolean("gameplay-mechanics.player.teleport-if-outside-border", teleportIfOutsideBorder);
         teleportOnNetherCeilingDamage = getBoolean("gameplay-mechanics.player.teleport-on-nether-ceiling-damage", teleportOnNetherCeilingDamage);
         totemOfUndyingWorksInInventory = getBoolean("gameplay-mechanics.player.totem-of-undying-works-in-inventory", totemOfUndyingWorksInInventory);
         playerFixStuckPortal = getBoolean("gameplay-mechanics.player.fix-stuck-in-portal", playerFixStuckPortal);
-        creativeOnePunch = getBoolean("gameplay-mechanics.player.one-punch-in-creative", creativeOnePunch);
         playerSleepNearMonsters = getBoolean("gameplay-mechanics.player.sleep-ignore-nearby-mobs", playerSleepNearMonsters);
         playersSkipNight = getBoolean("gameplay-mechanics.player.can-skip-night", playersSkipNight);
         playerCriticalDamageMultiplier = getDouble("gameplay-mechanics.player.critical-damage-multiplier", playerCriticalDamageMultiplier);
@@ -1240,11 +1209,6 @@ public class PurpurWorldConfig {
         spongeAbsorbsWaterFromMud = getBoolean("blocks.sponge.absorbs-water-from-mud", spongeAbsorbsWaterFromMud);
     }
 
-    public float stonecutterDamage = 0.0F;
-    private void stonecutterSettings() {
-        stonecutterDamage = (float) getDouble("blocks.stonecutter.damage", stonecutterDamage);
-    }
-
     public boolean turtleEggsBreakFromExpOrbs = false;
     public boolean turtleEggsBreakFromItems = false;
     public boolean turtleEggsBreakFromMinecarts = false;
