From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: OverwriteMC <artyom030803@gmail.com>
Date: Mon, 9 Feb 2026 12:37:26 +0300
Subject: [PATCH] Optimize map lookups with isEmpty check


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 9efb8061115a5c5a241a5361a8dbafb1b2ea8ab3..5003ae7560db038b96eac487c17a3a9b07f49edf 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -252,6 +252,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     private int id = ENTITY_COUNTER.incrementAndGet();
     public boolean blocksBuilding;
     public ImmutableList<Entity> passengers = ImmutableList.of();
+    public static final ImmutableList<Entity> EMPTY_PASSENGERS = ImmutableList.of(); // Leaf - Optimize map lookups with isEmpty check
     protected int boardingCooldown;
     private @Nullable Entity vehicle;
     private Level level;
@@ -478,7 +479,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
 
     @Override
     public final boolean moonrise$hasAnyPlayerPassengers() {
-        if (this.passengers.isEmpty()) {
+        if (this.passengers == EMPTY_PASSENGERS) { // Leaf - Optimize map lookups with isEmpty check
             return false;
         }
         return this.getIndirectPassengersStream().anyMatch((entity) -> entity instanceof Player);
@@ -3644,7 +3645,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
         if (passenger.getVehicle() != this) {
             throw new IllegalStateException("Use x.startRiding(y), not y.addPassenger(x)");
         } else {
-            if (this.passengers.isEmpty()) {
+            if (this.passengers == EMPTY_PASSENGERS) { // Leaf - Optimize map lookups with isEmpty check
                 this.passengers = ImmutableList.of(passenger);
             } else {
                 List<Entity> list = Lists.newArrayList(this.passengers);
@@ -3754,7 +3755,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     protected boolean canAddPassenger(Entity passenger) {
-        return this.passengers.isEmpty();
+        return this.passengers == EMPTY_PASSENGERS; // Leaf - Optimize map lookups with isEmpty check
     }
 
     protected boolean couldAcceptPassenger() {
@@ -3942,7 +3943,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     public boolean isVehicle() {
-        return !this.passengers.isEmpty();
+        return this.passengers != EMPTY_PASSENGERS; // Leaf - Optimize map lookups with isEmpty check
     }
 
     public boolean dismountsUnderwater() {
@@ -4950,11 +4951,11 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     public @Nullable Entity getFirstPassenger() {
-        return this.passengers.isEmpty() ? null : this.passengers.get(0);
+        return this.passengers == EMPTY_PASSENGERS ? null : this.passengers.get(0); // Leaf - Optimize map lookups with isEmpty check
     }
 
     public boolean hasPassenger(Entity entity) {
-        return this.passengers.contains(entity);
+        return passengers != EMPTY_PASSENGERS && this.passengers.contains(entity); // Leaf - Optimize map lookups with isEmpty check
     }
 
     public boolean hasPassenger(Predicate<Entity> predicate) {
@@ -4968,19 +4969,19 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     private Stream<Entity> getIndirectPassengersStream() {
-        if (this.passengers.isEmpty()) { return Stream.of(); } // Paper - Optimize indirect passenger iteration
+        if (this.passengers == EMPTY_PASSENGERS) { return Stream.of(); } // Paper - Optimize indirect passenger iteration // Leaf - Optimize map lookups with isEmpty check
         return this.passengers.stream().flatMap(Entity::getSelfAndPassengers);
     }
 
     @Override
     public Stream<Entity> getSelfAndPassengers() {
-        if (this.passengers.isEmpty()) { return Stream.of(this); } // Paper - Optimize indirect passenger iteration
+        if (this.passengers == EMPTY_PASSENGERS) { return Stream.of(this); } // Paper - Optimize indirect passenger iteration // Leaf - Optimize map lookups with isEmpty check
         return Stream.concat(Stream.of(this), this.getIndirectPassengersStream());
     }
 
     @Override
     public Stream<Entity> getPassengersAndSelf() {
-        if (this.passengers.isEmpty()) { return Stream.of(this); } // Paper - Optimize indirect passenger iteration
+        if (this.passengers == EMPTY_PASSENGERS) { return Stream.of(this); } // Paper - Optimize indirect passenger iteration // Leaf - Optimize map lookups with isEmpty check
         return Stream.concat(this.passengers.stream().flatMap(Entity::getPassengersAndSelf), Stream.of(this));
     }
 
@@ -4988,7 +4989,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
         // Paper start - optimise entity tracker
         final List<Entity> ret = new ArrayList<>();
 
-        if (this.passengers.isEmpty()) {
+        if (this.passengers == EMPTY_PASSENGERS) { // Leaf - Optimize map lookups with isEmpty check
             return ret;
         }
 
@@ -5003,7 +5004,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
     }
 
     public boolean hasExactlyOnePlayerPassenger() {
-        if (this.passengers.isEmpty()) { return false; } // Paper - Optimize indirect passenger iteration
+        if (this.passengers == EMPTY_PASSENGERS) { return false; } // Paper - Optimize indirect passenger iteration // Leaf - Optimize map lookups with isEmpty check
         return this.countPlayerPassengers() == 1;
     }
 
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index f5511d1951a3aff7e59ed90f2cd8773976f14067..5c7e307e59190154305dcb5a8cd71f92aceecdc8 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1216,7 +1216,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     }
 
     public boolean hasEffect(Holder<MobEffect> effect) {
-        return this.activeEffects.containsKey(effect);
+        return !activeEffects.isEmpty() && this.activeEffects.containsKey(effect); // Leaf - Optimize map lookups with isEmpty check
     }
 
     public @Nullable MobEffectInstance getEffect(Holder<MobEffect> effect) {
@@ -4330,7 +4330,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     // CraftBukkit start - collidable API
     @Override
     public boolean canCollideWithBukkit(Entity entity) {
-        return this.isPushable() && this.collides != this.collidableExemptions.contains(entity.getUUID());
+        return this.isPushable() && this.collides != (!this.collidableExemptions.isEmpty() && this.collidableExemptions.contains(entity.getUUID())); // Leaf - Optimize map lookups with isEmpty check
     }
     // CraftBukkit end
 
diff --git a/net/minecraft/world/entity/Mob.java b/net/minecraft/world/entity/Mob.java
index b6dbe774787be28c9fff7545eddaa9d24643503e..b60b5c57323be54e2d92c012fbb6e6e7a07c097c 100644
--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -1732,7 +1732,7 @@ public abstract class Mob extends LivingEntity implements EquipmentUser, Leashab
         if (!player.getItemInHand(hand).isEmpty()) {
             return InteractionResult.PASS;
         }
-        if (!passengers.isEmpty() || player.isPassenger()) {
+        if (passengers != Entity.EMPTY_PASSENGERS || player.isPassenger()) { // Leaf - Optimize map lookups with isEmpty check
             return InteractionResult.PASS;
         }
         if (this instanceof TamableAnimal tamable) {
diff --git a/net/minecraft/world/entity/monster/piglin/Piglin.java b/net/minecraft/world/entity/monster/piglin/Piglin.java
index 5e04a6792f26fad97b31f819fdf77b105c179564..aaee7eb3b05ef84ae1ff263d54296f03b477a544 100644
--- a/net/minecraft/world/entity/monster/piglin/Piglin.java
+++ b/net/minecraft/world/entity/monster/piglin/Piglin.java
@@ -529,7 +529,7 @@ public class Piglin extends AbstractPiglin implements CrossbowAttackMob, Invento
 
     private Entity getTopPassenger(Entity vehicle, int maxPosition) {
         List<Entity> passengers = vehicle.getPassengers();
-        return maxPosition != 1 && !passengers.isEmpty() ? this.getTopPassenger(passengers.getFirst(), maxPosition - 1) : vehicle;
+        return maxPosition != 1 && passengers != Entity.EMPTY_PASSENGERS ? this.getTopPassenger(passengers.getFirst(), maxPosition - 1) : vehicle; // Leaf - Optimize map lookups with isEmpty check
     }
 
     @Override
diff --git a/net/minecraft/world/entity/vehicle/minecart/AbstractMinecart.java b/net/minecraft/world/entity/vehicle/minecart/AbstractMinecart.java
index 48099ae66045b6e78ec52832e3b972ac9c9bc246..a297c683850464b9e4102b9db94040c0daae176d 100644
--- a/net/minecraft/world/entity/vehicle/minecart/AbstractMinecart.java
+++ b/net/minecraft/world/entity/vehicle/minecart/AbstractMinecart.java
@@ -434,7 +434,7 @@ public abstract class AbstractMinecart extends VehicleEntity {
         this.setDeltaMovement(Mth.clamp(deltaMovement.x, -maxSpeed, maxSpeed), deltaMovement.y, Mth.clamp(deltaMovement.z, -maxSpeed, maxSpeed));
 
         // Purpur start - Minecart settings and WASD controls
-        if (level().purpurConfig.minecartControllable && !isInWater() && !isInLava() && !passengers.isEmpty()) {
+        if (level().purpurConfig.minecartControllable && !isInWater() && !isInLava() && passengers != Entity.EMPTY_PASSENGERS) { // Leaf - Optimize map lookups with isEmpty check
             Entity passenger = passengers.get(0);
             if (passenger instanceof net.minecraft.server.level.ServerPlayer player) {
                 net.minecraft.world.entity.player.Input lastClientInput = player.getLastClientInput();
