From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lumine1909 <133463833+Lumine1909@users.noreply.github.com>
Date: Thu, 21 May 2026 14:05:19 +0800
Subject: [PATCH] Pre authenticate event


diff --git a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index 454662d862867be8d6019ed10142ec45b8bdd448..a2a8bdcd9fc3db2fe421e53fdd35bcde2121360a 100644
--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -149,7 +149,10 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
         this.requestedUuid = packet.profileId();
         // Paper end - Validate usernames
         this.requestedUsername = packet.name();
-        if (this.server.usesAuthentication() && !this.connection.isMemoryConnection()) {
+        // Leaf start - pre authenticate event
+        boolean shouldAuthenticate = new org.dreeam.leaf.event.AsyncPreAuthenticateEvent(requestedUsername, requestedUuid, connection.getRemoteAddress(), !server.usesAuthentication()).callEvent();
+        if (shouldAuthenticate && !this.connection.isMemoryConnection()) {
+            // Leaf end - pre authenticate event
             this.state = ServerLoginPacketListenerImpl.State.KEY;
             this.connection.send(new ClientboundHelloPacket("", this.server.getKeyPair().getPublic().getEncoded(), this.challenge, true));
         } else {
